File: C++/2/2.cpp
#include <iostream>

int even_fib_sum(int bound);

int main(int argc, char* argv[]) {
  std::cout << even_fib_sum(4e6) << '\n';
  return 0;
}

int even_fib_sum(int bound) {
  int first = 1, second = 2;
  // Sum so far is second because it is even
  int sum = second;
  for (int current = first + second; current < bound; current = first + second) {
    if (current % 2 == 0) {
      sum += current;
    }
    first = second;
    second = current;
  }
  return sum;;
}

File: Python/2/2.py
def even_fib(limit):
    first = 1
    second = 2
    sum = 2
    next = first + second
    while next <= limit:
        if next % 2 == 0:
            sum += next
        first = second
        second = next
        next = first + second
    return sum

print(even_fib(4e6))
